ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -678,24 +678,35 @@ let kdcs_of_domain domain =
        )
   with _ -> raise (generic_ex "Failed to lookup kdcs of domain %s" domain)
 
-let kdc_of_domain domain =
-  let msg = Printf.sprintf "Failed to lookup kdc of domain %s" domain in
-  kdcs_of_domain domain |> hd msg
+let workgroup_from_server kdc =
+  let err_msg =
+    Printf.sprintf "Failed to lookup workgroup from server %s" kdc
+  in
+  let key = "Pre-Win2k Domain" in
+  try
+    Helpers.call_script ~log_output:On_failure net_cmd
+      ["ads"; "lookup"; "-S"; kdc; "-d"; debug_level ()]
+    |> fun lines -> Xapi_cmd_result.of_output ~sep:':' ~key lines
+  with _ -> raise (Auth_service_error (E_LOOKUP, err_msg))
```

Comment:
Should we log what the actual error was here, at least at debug level?

Reply:
And instead of raising an exception here it might be better to return `Error (Auth_service_error (E_LOOKUP, err_msg))`, and an `Ok ()` on the `try` side if it succeeds? Then you can use this as a predicate in a find_opt if you transform the result into a bool